app/models/lit/base.rb

```diff
@@ -12,30 +14,25 @@ def mark_for_retry_on_update
     @will_retry_update = true
   end
 
-
   after_rollback :retry_lit_model_save
 
   private
 
   def retry_lit_model_save
-    retry_on_create if @will_retry_create
-    retry_on_update if @will_retry_update
+    retry_on_create if instance_variable_defined?(:@will_retry_create) && @will_retry_create
+    retry_on_update if instance_variable_defined?(:@will_retry_update) && @will_retry_update
   end
 
   def retry_on_create
-    return if @retry_created
-    @retry_created = true
-    self.class.create attributes
+    return if self.retried_created
+    self.retried_created = true
+    self.class.create! attributes.merge(retried_created: true)
   end
 
   def retry_on_update
-    return if @retry_updated
-    @retry_updated = true
-    update attributes
-  end
-
-  def lit_attribute_will_change
-    raise NotImplementedErrror
+    return if self.retried_updated
+    self.retried_updated = true
+    update attributes.merge(retried_updated: true)
```

Comment:
Can we use `update!` here as well as in the `retry_on_create` method so that these methods behave consistently?